### PR TITLE
Azure.Identity adding KnownAuthorityHosts constants

### LINKED ISSUE #10228: [FEATURE REQ] Azure.Identity - Support KnownAuthorities like Python SDK
Azure.Identity

It would be great if we provided a "KnownAuthorities" concept, like we have in the Python SDK.
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/azure/identity/_constants.py

Right now, in .NET, devs have to provide the url for the environment they are targetting `https://login.microsoftonline.us` for Gov, for example.

With Python
```python
credential = DefaultAzureCredential(authority=KnownAuthorities.AZURE_GOVERNMENT)
```

With .NET
```csharp
var options = new DefaultAzureCredentialOptions { AuthorityHost = new Uri("https://login.microsoftonline.us/") };
var client = new KeyClient(new Uri(keyVaultUrl), new DefaultAzureCredential(options));
```

### PR DESCRIPTION
**Pull request by schaabs**
This pull request adds a static class `KnownAuthorityHosts` which defines fields for the authority host Uri's for all known azure cloud instances. 

Fixes #10228

### CODE REVIEW

File: sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs [hunk 1]
```diff
@@ -7,14 +7,12 @@
...
-    /// Options to configure requests made to the OAUTH identity service
+    /// Options to configure requests made to the OAUTH identity service.
     /// </summary>
     public class TokenCredentialOptions : ClientOptions
     {
-        private static readonly Uri s_defaultAuthorityHost = new Uri("https://login.microsoftonline.com/");
-
         /// <summary>
-        /// The host of the Azure Active Directory authority.   The default is https://login.microsoftonline.com/
+        /// The host of the Azure Active Directory authority. The default is https://login.microsoftonline.com/. For common authority hosts for known azure cloud instances see <see cref="KnownAuthorityHosts"/>.
```
Comment (jongio) [inline sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs]: Uppercase A on Azure.

Comment (jongio) [inline sdk/identity/Azure.Identity/src/TokenCredentialOptions.cs]: Here you put common, above you use 'well known'. Would be good to go with one or the other.

**Review by jongio — APPROVED** [review]
Thank you.

File: sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs [hunk 2]
```diff
@@ -97,6 +97,13 @@ public partial class InteractiveBrowserCredential : Azure.Core.TokenCredential
         public override Azure.Core.AccessToken GetToken(Azure.Core.TokenRequestContext requestContext, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public override System.Threading.Tasks.ValueTask<Azure.Core.AccessToken> GetTokenAsync(Azure.Core.TokenRequestContext requestContext, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public static partial class KnownAuthorityHosts
+    {
+        public static readonly System.Uri AzureChinaCloud;
+        public static readonly System.Uri AzureCloud;
+        public static readonly System.Uri AzureGermanCloud;
+        public static readonly System.Uri AzureUsGovernment;
```
Comment (jongio) [inline sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs]: Should Us be US?

File: sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs [hunk 3]
```diff
@@ -97,6 +97,13 @@ public partial class InteractiveBrowserCredential : Azure.Core.TokenCredential
         public override Azure.Core.AccessToken GetToken(Azure.Core.TokenRequestContext requestContext, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
         public override System.Threading.Tasks.ValueTask<Azure.Core.AccessToken> GetTokenAsync(Azure.Core.TokenRequestContext requestContext, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public static partial class KnownAuthorityHosts
```
Comment (jongio) [inline sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs]: On the name, python has KnownAuthorities and .NET has KnownAuthorityHosts.  I like the .NET version better, but should it be consistent with other langs?

Comment (schaabs) [reply sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs]: I intentionally left this different, as the inconsistency already exists. The option names are different. In python the kwarg which configures this is `authority`
~~~ python
credential = DefaultAzureCredential(authority=KnownAuthorities.AZURE_CHINA)
~~~
whereas in .NET it's `AuthorityHost`
~~~ c#
var options = new DefaultAzureCredentialOptions { AuthorityHost = KnownAuthorityHosts.AzureChinaCloud };

var client = new KeyClient(new Uri(keyVaultUrl), new DefaultAzureCredential(options));
~~~

Given that we can't change the property name on the options, I think it's more important that the class match this than the class name in a different language.

Comment (jongio) [reply sdk/identity/Azure.Identity/api/Azure.Identity.netstandard2.0.cs]: Sounds good.  Thanks

**Pull request merged**